mongo/integration/cursor_test.go

```diff
@@ -165,6 +165,33 @@ func TestCursor(t *testing.T) {
 			assertCursorBatchLength(mt, cursor, len(getMoreBatch)-1)
 		})
 	})
+	mt.RunOpts("all", noClientOpts, func(mt *mtest.T) {
+		failpointOpts := mtest.NewOptions().Topologies(mtest.ReplicaSet).MinServerVersion("4.0")
+		mt.RunOpts("getMore error", failpointOpts, func(mt *mtest.T) {
+			failpointData := mtest.FailPointData{
+				FailCommands: []string{"getMore"},
+				ErrorCode:    100,
+			}
+			mt.SetFailPoint(mtest.FailPoint{
+				ConfigureFailPoint: "failCommand",
+				Mode:               "alwaysOn",
+				Data:               failpointData,
+			})
+			initCollection(mt, mt.Coll)
+			cursor, err := mt.Coll.Find(mtest.Background, bson.D{}, options.Find().SetBatchSize(2))
+			assert.Nil(mt, err, "Find error: %v", err)
+			defer cursor.Close(mtest.Background)
+
+			var docs []bson.D
+			err = cursor.All(context.Background(), &docs)
+			assert.NotNil(mt, err, "expected change stream error, got nil")
+
+			// make sure that a mongo.CommandError is returned instead of a driver.Error
+			mongoErr, ok := err.(mongo.CommandError)
+			assert.True(mt, ok, "expected mongo.CommandError, got: %T", err)
+			assert.Equal(mt, failpointData.ErrorCode, mongoErr.Code, "expected code %v, got: %v", failpointData.ErrorCode, mongoErr.Code)
```

Comment:
Is it also worth checking `errors.As(err, driver.Error)` to make an assertion on the wrapped error?